cluster/cluster_test.go

```diff
@@ -19,9 +39,103 @@ func TestJoin(t *testing.T) {
 	require.NoError(t, err)
 	defer c.Close()
 
-	t.Run("test registery contains expected services", func(t *testing.T) {
+	t.Run("test registry contains expected services", func(t *testing.T) {
 		services, err := c.Registry.Services(ctx)
 		require.NoError(t, err)
 		require.NotEmpty(t, services["ping"])
 	})
 }
+
+func (suite *ClusterSuite) TestMemberAdd() {
+	t := suite.T()
+
+	cfg, err := ConfigFromFile("./testdata/ping.yml")
+	require.NoError(t, err)
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	c, err := Join(ctx, cfg)
+	require.NoError(t, err)
+	defer c.Close()
+
+	t.Run("test member list contains expected number of servers", func(t *testing.T) {
+		members, err := c.MemberList(ctx)
+		require.NoError(t, err)
+		require.Equal(t, 1, len(members))
+	})
+
+	t.Run("test member add successfully adds new member", func(t *testing.T) {
+		lcfg, err := embed.ConfigFromFile(fmt.Sprintf("./testdata/%s", cfg.EtcdConfigFile))
+		require.NoError(t, err)
+
+		LPUrl, err := url.Parse("http://127.0.0.1:22380")
+		require.NoError(t, err)
+
+		LCUrl, err := url.Parse("http://127.0.0.1:22379")
+		require.NoError(t, err)
+
+		APUrl, err := url.Parse("http://127.0.0.1:22380")
+		require.NoError(t, err)
+
+		ACUrl, err := url.Parse("http://127.0.0.1:22379")
+		require.NoError(t, err)
+
+		mecfg := &EtcdConfig{
+			Name:           "node2",
+			DataDir:        "tmp2",
+			LPUrls:         []url.URL{*LPUrl},
+			LCUrls:         []url.URL{*LCUrl},
+			APUrls:         []url.URL{*APUrl},
+			ACUrls:         []url.URL{*ACUrl},
+			InitialCluster: "node2=http://127.0.0.1:22380",
+		}
+
+		mcfg := embed.NewConfig()
+		mcfg.Name = mecfg.Name
+		mcfg.Dir = mecfg.DataDir
+		mcfg.LPUrls = mecfg.LPUrls
+		mcfg.APUrls = mecfg.APUrls
+		mcfg.ACUrls = mecfg.ACUrls
```

Comment:
a lot of this logic will hopefully be moved out once i figure out part 2 😓 did not want to create another `.yml` file just to make sure the caller can create the etcd server correctly